crates/sui-adapter/src/execution_engine.rs

```diff
@@ -342,11 +332,11 @@ fn execute_transaction<
                   // check conservation once more more. if we still fail, it's a problem with gas
                   // charging that happens even in the "aborted" case--no other option but panic.
                   // we will create or destroy SUI otherwise
-                  temporary_store.check_sui_conserved(advance_epoch_gas_summary, do_expensive_checks).unwrap();
+                  temporary_store.check_sui_conserved(advance_epoch_gas_summary, enable_expensive_checks).unwrap();
```

Comment:
can we panic with a bit more info? I am not sure how we deal with panic off `unwrap` but if we go blind that is not a good thing. Particularly the `tx_digest` would be really good to have so the panic shows that info right away